Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/TextSolutionFileParser.cs

```diff
@@ -0,0 +1,395 @@
+    {
+        PreSolution,
+        PostSolution
+    }
+
+    internal class SolutionFileInfo
+    {
+        public Dictionary<Guid, Project> Projects { get; set; }
+
+        public Dictionary<string, SolutionFileSection<SolutionGlobalSectionType>> SolutionSections { get; set; }
+
+        public Dictionary<Guid, Guid> ChildToParentNestedMappings { get; set; }
+
+        public HashSet<ConfigPlatformPair> ConfigPlatformPairs { get; set; }
+
+        public Dictionary<string, string> Properties { get; set; }
+
+        public Dictionary<string, string> ExtensibilityGlobals { get; set; }
+
+        public Dictionary<string, string> SolutionNotes { get; set; }
+
+        public Dictionary<Guid, List<ProjectConfigurationEntry>> ProjectConfigurationEntires { get; set; }
+
+        public HashSet<Guid> MSB4UGeneratedItems { get; set; }
+    }
+
+    internal struct ConfigPlatformPair
+    {
+        public string Configuration { get; set; }
+
+        public string Platform { get; set; }
+
+        public ConfigPlatformPair(string configuration, string platform)
+        {
+            Configuration = configuration;
+            Platform = platform;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is ConfigPlatformPair other
+                && Equals(Configuration, other.Configuration)
+                && Equals(Platform, other.Platform);
+        }
+
+        public override int GetHashCode()
+        {
+            return (Configuration?.GetHashCode() ?? 0)
+                ^ (Platform?.GetHashCode() ?? 0);
+        }
+
+        internal struct Comparer : IComparer<ConfigPlatformPair>
+        {
+            internal static Comparer Instance { get; } = new Comparer();
+
+            public int Compare(ConfigPlatformPair x, ConfigPlatformPair y)
+            {
+                int results = string.Compare(x.Configuration, y.Configuration);
+
+                return results == 0
+                    ? string.Compare(x.Platform, y.Platform)
+                    : results;
+            }
+        }
+    }
+
+    internal struct ProjectConfigurationEntry
+    {
+        public ConfigPlatformPair SolutionConfig { get; set; }
+
+        public ConfigPlatformPair ProjectConfig { get; set; }
+
+        public string Property { get; set; }
+    }
+
+    internal struct SolutionFileSection<T>
+    {
+        public string Name { get; set; }
+
+        public T Type { get; set; }
+
+        public List<string> Lines { get; set; }
+    }
+
+    internal struct Project
+    {
+        public Guid TypeGuid { get; set; }
+
+        public string Name { get; set; }
+
+        public string RelativePath { get; set; }
+
+        public Guid Guid { get; set; }
+
+        public HashSet<Guid> Dependencies { get; set; }
+
+        public Dictionary<string, SolutionFileSection<SolutionProjecSectionType>> Sections { get; set; }
+    }
+
+    internal static class TextSolutionFileParser
```

Comment:
>TextSolutionFileParser [](start = 26, length = 22)

I haven't looked much in to it, but just making sure you're aware of the Microsoft functionality around solution/project parsing.

[https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2013/dn763266(v=vs.121](https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2013/dn763266(v=vs.121))

It looks like it *may* have been moved to be internal-only functionality still accessible through reflection, so possibly not the best direction, but just making sure it is known about.

Reply:
Yes, here is the problem, I need to reference a NuGet package for that. This presents a form of Catch 22.

---
In reply to: [364913591](https://github.com/microsoft/MSBuildForUnity/pull/96#discussion_r364913591) [](ancestors = 364913591)